src/content/docs/react/hooks.mdx

```diff
@@ -1,3 +1,412 @@
+    return someExpensiveCalculation();
+});
+```
+
+### State Updation
+
+When updating state based on the current state (e.g., incrementing or toggling values), you can pass a function to setState. This ensures you’re working with the latest state value.
+
+```jsx
+<button onClick={() => setCount(prevCount => prevCount + 1)}>
+    Increment
+</button>
+```
+
+### `useEffect` : Handling Side effects
+
+The `useEffect` hook is used to handle side effects in functional components. Common side effects include data fetching, subscriptions, and directly interacting with the DOM. Before hooks, these kinds of side effects were managed using lifecycle methods like `componentDidMount`, `componentDidUpdate`, and `componentWillUnmount` in class components. `useEffect` combines all of these into one unified API. Basically, it takes two arguments
+ - A function where you define the side effect.
+ - An optional dependency array that controls when the effect should run.
+
+**Syntax of `useEffect`**
+
+```jsx
+useEffect(() => {
+    // Code for the side effect
+}, [dependencies]);
+```
+
+Side effect function is the function that runs after each render, performing the desired effect (e.g., fetching data).
+Dependency is an optional array that specifies when the effect should re-run. If a value in the array changes, the effect will re-run.
+
+### Example of `useEffect`
+
+import DataFetchingComponent from './codes/hooksCode/useEffect/DataFetchingComponent.txt?raw';
+import useEffectUsage from './codes/hooksCode/useEffect/Usage.txt?raw';
+
+<Tabs>
+  <TabItem label="Dependencies">
+    Install the following libraries. Assuming you have installed
+    <Code code={`npm install react react-dom`} lang="shell" />
+  </TabItem>
+  <TabItem label="Usage">
+    <Code code={useEffectUsage} lang="jsx" title="Usage.jsx" />
+  </TabItem>
+  <TabItem label="Component">
+    <Code code={DataFetchingComponent} lang="jsx" title="DataFetchingComponent.jsx" />
+  </TabItem>
+</Tabs>
+
+In this example: The side effect is fetching data from an API when the component first renders. The [] (empty dependency array) ensures the effect only runs once, similar to `componentDidMount`.
+
+### Controlling using Dependency array
+
+You can control when an effect runs by adding dependencies. The effect will only run again if one of the dependencies changes.
+
+```jsx
+useEffect(() => {
+    // This effect runs when 'count' changes
+    console.log(`The count is now ${count}`);
+}, [count]); // Effect runs only when 'count' changes
+```
+
+### Cleaning up effects
+
+Sometimes, side effects like subscriptions or timers need to be cleaned up to prevent memory leaks. `useEffect` can return a cleanup function that runs when the component unmounts or before the effect re-runs.
+
+import Cleanup from './codes/hooksCode/Others/Cleanup.txt?raw';
+
+<Tabs>
+  <TabItem label="Usage">
+    <Code code={Cleanup} lang="jsx" title="CleanUp.jsx" />
+  </TabItem>
+</Tabs>
+
+### Additional Hooks
```

Comment:
Please make sure to use different heading methods to differentiate between main headings and sub headings. All headings looks same and it ain't looking good overall.